src/linalg.jl

```diff
@@ -1,8 +1,15 @@
-function mulαβ!(C::Matrix{T}, A::Matrix{T}, adjB::Adjoint{T,<:Matrix{T}}, 
+function mulαβ!(C::Matrix{T}, A::Matrix{T}, adjB::Adjoint{T,<:Matrix{T}},
         α=true, β=false) where {T<:BlasFloat}
     BLAS.gemm!('N', 'C', T(α), A, adjB.parent, T(β), C)
 end
 
+function mulαβ!(C::Matrix{T}, A::Matrix{T}, adjB::Diagonal{T,S},
+        α=true, β=false) where {T<:BlasFloat,S}
+    # this could be more efficient for the diagonal case
+    BLAS.gemm!('N', 'C', T(α), A, Matrix(adjB), T(β), C)
+end
+
```

Comment:
These methods will no longer be needed after v1.3.0 is released.  All of the `mulαβ!` methods will become `mul!` methods and only the ones that involve matrix types defined in this package will need to be defined.  The version in LinearAlgebra/src/diagonal.jl for julia 1.3.0 is
```julia
@inline mul!(out::AbstractMatrix, in::StridedMatrix, A::Diagonal,
             alpha::Number, beta::Number) =
    out .= (in .* permutedims(A.diag)) .*ₛ alpha .+ out .*ₛ beta
```

Reply:
How do we want to handle the transition to 1.3? Just drop these methods and support for <1.3 ?

Reply:
Yes, that would be my preferred approach.